### PR TITLE
Change FID to run per epoch and for a specified num_batches

### PR DESCRIPTION
**Pull request by jeswan**
This PR

1. Changes the default device for FID to cuda
2. Creates the inception model once instead of every `calculate_fid()`
3. Shuffles the  validation dataloader for randomized fid calculations
4. Changes FID to run per num_epochs and for num_batches

### CODE REVIEW

File: src/metrics/fid.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from pytorch_fid.inception import InceptionV3
 from pytorch_fid.fid_score import calculate_frechet_distance
 
-def get_activations(batch, model, dims=2048, device='cpu'):
+def get_activations(batch, model, dims=2048, device='gpu'):
```
Comment (jeswan) [inline src/metrics/fid.py:9]:
```suggestion
def get_activations(batch, model, dims=2048, device='cuda'):
```

File: src/metrics/fid.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def get_activations(batch, model, dims=2048, device='cpu'):
 
     return pred_arr
 
-def calculate_activation_statistics(batch, model, dims=2048, device='cpu'):
+def calculate_activation_statistics(batch, model, dims=2048, device='gpu'):
```
Comment (jeswan) [inline src/metrics/fid.py:50]:
```suggestion
def calculate_activation_statistics(batch, model, dims=2048, device='cuda'):
```

File: src/metrics/fid.py [hunk 3]
```diff
@@ -68,20 +68,22 @@ def calculate_activation_statistics(batch, model, dims=2048, device='cpu'):
     sigma = np.cov(act, rowvar=False)
     return mu, sigma
 
-def calculate_fid(preds, truths, is_color=False, dims=2048, device='cpu'):
+def calculate_fid(preds, truths, is_color=False, dims=2048, device='gpu', inception_model=None):
```
Comment (jeswan) [inline src/metrics/fid.py:71]:
```suggestion
def calculate_fid(preds, truths, is_color=False, dims=2048, device='cuda', inception_model=None):
```

File: src/data/data.py [hunk 4]
```diff
@@ -268,7 +268,7 @@ def train_dataloader(self):
     def val_dataloader(self):
         loader = DataLoader(self.val_dataset,
                             batch_size=self.batch_size,
-                            shuffle=False,
+                            shuffle=True,
```
Comment (annikabrundyn) [inline src/data/data.py:271]: we def shouldnt set shuffle to true in val dataloader

Comment (annikabrundyn) [reply src/data/data.py:271]: i tried to do this by picking a random batch id in the other pr but its def not advised to shuffle val batches - any other ideas? :)

Comment (jeswan) [reply src/data/data.py:271]: random batch id is an alternative. are you worried about floating point errors when doing val shuffling? shuffling should be largely irrelevant

Comment (annikabrundyn) [reply src/data/data.py:271]: its just not good practice to shuffle your val dataset - we want this to be consistent across different models etc. i actually think if you run this there is a warning in lightning tell you not to do this

Comment (jeswan) [reply src/data/data.py:271]: fair enough :) will switch to a random batch.

**Comment by annikabrundyn** [comment]
@jeswan why not use the new fid callback in metrics?
